ossdev/__init__.py

```diff
@@ -21,51 +24,178 @@ def set(self, arr):
     def get(self):
         return self.d
 
+    def clone(self):
+        return Vector.from_arr(list(self.get()))
+
+    @property
+    def is_row(self):
+        return self.row
+
     def __len__(self):
         return len(self.d)
 
     def __repr__(self):
         return str(self.d)
 
+    def __str__(self):
+        return 'Vct%s' % self.d
+
     def __getitem__(self, item):
         return self.d[item]
 
     def __hash__(self):
         return sum(self.d)
 
     def __setitem__(self, key, value):
-        # TODO: implement
-        return None
+        self.d[key] = value
+
+    # magic __cmp__ function is deprecated since Python3
+    # def __cmp__(self, other):
+    #     return (self.length() > other.length()) - (self.length() < other.length())
+
+    def __eq__(self, other):
+        return self.length() == other.length()
```

Comment:
comparison just by length is not enough!

Reply:
You dont have to fix it. Just a note.

Reply:
If you choose to remove `__cmp__` you should properly implement other comparators. 
E.g., with https://docs.python.org/3/library/functools.html#functools.total_ordering